contributed_definitions/nyaml/NXxps.yaml

```diff
@@ -0,0 +1,451 @@
+            * hard X-ray photoemission spectroscopy (HAXPES)
+            * near ambient pressure X-ray photoelectron spectroscopy (NAPXPS)
+            * electron spectroscopy for chemical analysis (ESCA)
+        
+        .. _ISO 18115-1:2023: https://www.iso.org/standard/74811.html
+        .. _Clause 11: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:sec:11
+    xps_coordinate_system(NXcoordinate_system):
+      doc: |
+        Description of the XPS base coordinate system, which is defined such that the positive z-axis
+        points along the sample normal, and the x- and y-axes lie within the sample surface.
+        
+        The fields `x`, `y`, `z`, and `scaling` are used to describe the orientation of the XPS
+        coordinate system with respect to the beam coordinate system (.).
+      origin:
+        enumeration: [sample surface]
+      handedness:
+        enumeration: [right_handed]
+      z_direction:
+        enumeration: [sample normal]
+      x(NX_NUMBER):
+      y(NX_NUMBER):
+      z(NX_NUMBER):
+      offset(NX_NUMBER):
+    (NXinstrument):
+      doc: |
+        XPS spectrometer.
+      (NXelectronanalyser):
+        depends_on:
+          exists: recommended
+          doc: |
+            Reference to the transformation describing the orientation of the analyzer
+            relative to the beam.
+          enumeration: [transformations/angle_beam_analyser]
+        transformations(NXtransformations):
+          angle_beam_analyser(NX_NUMBER):
+            exists: recommended
+            unit: NX_ANGLE
+            doc: |
+              Angle between incoming beam and the center analyser axis.
+            \@transformation_type:
+              enumeration: [rotation]
+            \@depends_on:
+              doc: |
+                Should be a link to the XPS base coordinate system, e.g.
+                /entry/instrument/xps_coordinate_system.
+      (NXenergydispersion):
+        scheme:
+          enumeration: [hemispherical, double hemispherical, cylindrical mirror]
+        energy_scan_mode:
+          enumeration: [fixed_analyser_transmission, fixed_retardation_ratio, fixed_energy, snapshot]
+        diameter(NX_FLOAT):
+          exists: recommended
+    (NXsample):
+      depends_on(NX_CHAR):
+        exists: recommended
+        doc: |
+          Reference to the transformation describing the orientation of the sample
+          relative to the beam.
+        enumeration: [transformations/angle_beam_sample]
+      transformations(NXtransformations):
+        angle_beam_sample(NX_NUMBER):
+          exists: recommended
+          unit: NX_ANGLE
+          doc: |
+            Angle between incoming beam and the sample.
+          \@transformation_type:
+            enumeration: [rotation]
+          \@depends_on:
+            doc: |
+              Should be a link to the XPS base coordinate system, e.g.
+              /entry/instrument/xps_coordinate_system.
+    peak_fitting(NXpeak_model):
```

Comment:
I suggest unnamed instance of NXfit, to allow multiple fits.

Reply:
see below